### PR TITLE
Heaviside: use None to specify default 2nd-argument behaviour (alternative)

### PR DESCRIPTION
**Pull request by cbm755**
Maybe this is more palatable than #11243?

CC @zanzibar7

### CODE REVIEW

File: sympy/functions/special/delta_functions.py [hunk 1]
```diff
@@ -349,9 +349,29 @@ class Heaviside(Function):
                         ``( 0, if x < 0``
     2) ``Heaviside(x) = < ( undefined if x==0 [*]``
                         ``( 1, if x > 0``
+    3) ``Max(0,x).diff(x) = Heaviside(x)``
 
     .. [*] Regarding to the value at 0, Mathematica defines ``H(0) = 1``,
-           but Maple uses ``H(0) = undefined``
+           but Maple uses ``H(0) = undefined``.  Different application areas
+           may have specific convensions.  For example, in control theory, it
```
Comment (jksuom) [inline sympy/functions/special/delta_functions.py]: convensions -> conventions

**Comment by jksuom** [comment]
This looks good to me. +1

**Comment by cbm755** [comment]
Thanks!  @zanzibar7 are you ok with this?  (If not, let's merge your PR first and then discuss here.)

**Comment by zanzibar7** [comment]
This is nicer.  +1

**Comment by cbm755** [comment]
Thank you both, and especially @zanzibar7 as this is really his fix.

I'll merge in 24 hours, unless I hear otherwise.

**Comment by zanzibar7** [comment]
All good?  Once this is in, I'll revisit my proposal for handling Heaviside functions in Laplace transforms, #10955.

**Comment by BrodaUa** [comment]
Hello. 
This small uncertainty with Heaviside(0) brings pain to writing code. Can you please provide a way of evaluating the value of expression comprised with Heaviside(0)? For instance, if I got this: `1.0*Heaviside(0) + 1.0` or even more complex expr as result of derivation. Dealing with term `Heaviside(0)` alone quite easy via `eval()`, but about my case? 
Thanks.

**Pull request merged**